content/docs/pipelines/sdk/component-development.md

```diff
@@ -1,151 +1,67 @@
-  program.py --train-gcs-uri gs://bucket/datasets/1/train.tsv  
-  program.py --big-query-table my_table
-  ``` 
+When creating a component you need to think about how the component
+will communicate with upstream and downstream components. That is, how it will consume
+the input data and produce the output data.
 
-### More about output data
+### Producing data
 
-The program must write the output data to some location and inform the system
+To output any piece of data, the component program must write the output data to some location and inform the system
 about that location so that the system can pass the data between steps.
-You should provide the paths to your output data as command-line arguments.
-That is, you should not hardcode the paths. 
-
-You can choose a suitable storage solution for your output data. Options include 
-the following:
-
-* [Google Cloud Storage](https://cloud.google.com/storage/docs/) is the
-  recommended default storage solution for writing output.
-* For structured data you can use 
-  [BigQuery](https://cloud.google.com/bigquery/docs/).
-  You must provide the specific URI/path or table name to which to write the
-  results.
-
-The program should do the following:
-
-* Upload the data to your chosen storage system.
-* Pass out a URI pointing to the data, by writing that URI to a file and 
-  instructing the system to pick it up and treat it as the value of a particular 
-  component output. 
+The program should accept the paths for the output data as command-line arguments. That is, you should not hardcode the paths.
 
-Note that the example below accepts both a URI for uploading the data into, and
-a file path to write that URI to.
+#### Advanced: Producing data in an external system
 
-```
-program.py --out-model-uri gs://bucket/163/output_model \
-           --out-model-uri-file /outputs/output_model_uri/data
-```
-
-Why should the program output the URI it has just received as an input argument?
-The reason is that the URIs specified in the pipeline are usually not the real
-URIs, but rather URI templates containing UIDs. The system resolves the URIs at
-runtime when the containerized program starts. Only the containerized program 
-sees the fully-resolved URI.
-
-Below is an example of such a URI:
+In some scenarios, the goal of the component is to create some data object in an external service (E.g. a BigQuery table).
+In thais case the component program should do that and then output some identifier of the produced data location (e.g. BigQuery table name) instead of the data itself.
+This scenario should be limited to cases where the data must be put into external system instead of keeping it inside the Pipelines system.
+The class of components where this behavior is common is exporters (e.g. "upload data to GCS").
+Note that the Pipelines system cannot provide consistency and reproducibility guarantees for the data outside of its control.
```

Comment:
s/Pipelines/Kubeflow Pipelines

Or just "pipelines"

Here and below.

Reason: We should use either the brand name, capitalized, or a generic name not capitalized. This provides clarity in the docs.

Reply:
Done